presto-parquet/src/main/java/io/prestosql/parquet/writer/ParquetWriters.java

```diff
@@ -174,8 +175,21 @@ private static PrimitiveValueWriter getValueWriter(ValuesWriter valuesWriter, Ty
         if (DATE.equals(type)) {
             return new DateValueWriter(valuesWriter, parquetType);
         }
-        if (TIMESTAMP.equals(type)) {
-            return new TimestampValueWriter(valuesWriter, type, parquetType);
+        if (TIME_MICROS.equals(type)) {
+            verifyParquetType(type, parquetType, OriginalType.TIME_MICROS);
+            return new TimeMicrosValueWriter(valuesWriter, type, parquetType);
+        }
+        if (TIMESTAMP_MILLIS.equals(type)) {
+            verifyParquetType(type, parquetType, OriginalType.TIMESTAMP_MILLIS);
+            return new BigintValueWriter(valuesWriter, type, parquetType);
+        }
+        if (TIMESTAMP_MICROS.equals(type)) {
+            verifyParquetType(type, parquetType, OriginalType.TIMESTAMP_MICROS);
+            return new BigintValueWriter(valuesWriter, type, parquetType);
+        }
+        if (TIMESTAMP_TZ_MICROS.equals(type)) {
```

Comment:
Is there no `TIMESTAMP_TZ_MILLIS`?

Reply:
I didn't bother since we won't need that until we support `timestamp with time zone` in Hive, but I can add that now.